### PR TITLE
add semimetrizable theorems

### LINKED ISSUE #626: Eliminating unused properties
As came up in #618 and #619, there are many properties in the database which are lacking any theorems. Their presence makes improving the database harder, as they currently clutter up the list for no value. The following is a complete list of them:
- [ ] [P83](https://topology.pi-base.org/properties/P000083) 	Almost Čech Complete
- [x] [P85](https://topology.pi-base.org/properties/P000085) 	Ascoli
- [ ] [P92](https://topology.pi-base.org/properties/P000092) 	Moving Off Property
- [ ] [P94](https://topology.pi-base.org/properties/P000094) 	q Space
- [ ] [P95](https://topology.pi-base.org/properties/P000095) 	I-tactic Banach-Mazur
- [ ] [P96](https://topology.pi-base.org/properties/P000096) 	II-tactic Banach-Mazur
- [x] [P102](https://topology.pi-base.org/properties/P000102) 	Semimetrizable
- [ ] [P104](https://topology.pi-base.org/properties/P000104) 	K Analytic
- [ ] [P105](https://topology.pi-base.org/properties/P000105) 	Angelic
- [ ] [P106](https://topology.pi-base.org/properties/P000106) 	Strictly Angelic
- [ ] [P107](https://topology.pi-base.org/properties/P000107) 	Pointwise Countable Type
- [ ] [P108](https://topology.pi-base.org/properties/P000108) 	Locally Čech Complete
- [ ] [P109](https://topology.pi-base.org/properties/P000109) 	Countable Type
- [ ] [P110](https://topology.pi-base.org/properties/P000110) 	Has A Compact Resolution
- [ ] [P113](https://topology.pi-base.org/properties/P000113) 	kℝ Space
- [ ] [P115](https://topology.pi-base.org/properties/P000115) 	Weakly K Analytic
- [ ] [P117](https://topology.pi-base.org/properties/P000117) 	M Space
- [ ] [P118](https://topology.pi-base.org/properties/P000118) 	Pseudo-Polish
- [ ] [P119](https://topology.pi-base.org/properties/P000119) 	Z-Compact
- [ ] [P120](https://topology.pi-base.org/properties/P000120) 	r Space
- [ ] [P122](https://topology.pi-base.org/properties/P000122) 	S space

The two approaches we can take for each are to either delete them or to properly flesh them out. The only one of particular interest to me is semimetrizability, but I felt I should share the list for discussion if any items are of interest to others.

### PR DESCRIPTION
**Pull request by StevenClontz**
Closes #626.

@danflapjax probably has much more to add, but I wanted to go ahead and get this wired up with the rest of pi-Base, and this feels like the MVP to do so.

### CODE REVIEW

**Comment by prabau** [comment]
I know this is in progress.  But I hope you can find something better than Heath for the definition.  Compare with Encyclopedia of General Topology and other sources.

**Comment by StevenClontz** [comment]
> I know this is in progress. But I hope you can find something better than Heath for the definition. Compare with Encyclopedia of General Topology and other sources.

Certainly I could, but it's a strict improvement on the status quo, and *Procedings of the AMS* ain't chopped liver. If that's the only issue, maybe merge and open an issue?

**Comment by StevenClontz** [comment]
(I know Bob Heath - he was a regular participant of Auburn's topology seminar, so I'm a bit biased!)

**Comment by david20000813** [comment]
Since we’re discussing semimetrizable spaces and no one has yet brought this up, I’ve noticed that there are two inequivalent definitions of semimetrizability in literature:

1. A topological space $X$ is called semimetrizable iff there exists a semimetric $d$ on $X$ s.t. a subset $A \subset X$ is open iff for every $a \in A$, there exists $r > 0$ s.t. $\{x \in X: d(a, x) < r\} \subset A$ (equivalently, a subset $A \subset X$ is closed iff for every sequence $a_n \in A$ and element $p \in X$ s.t. $d(a_n, p) \to 0$, we have $p \in A$);
2. A topological space $X$ is called semimetrizable iff there exists a semimetric $d$ on $X$ s.t. when $A \subset X$, we have $p \in \overline{A}$ iff there exists a sequence $a_n \in A$ s.t. $d(a_n, p) \to 0$.

(In both cases, the definition of a semimetric is the same: it is a function $d: X \times X \to \mathbb{R}_{\geq 0}$ s.t. $d(x, y) = 0$ iff $x = y$ and $d(x, y) = d(y, x)$ for all $x, y \in X$.)

The first definition appears, for example, in Wilson’s paper: https://www.jstor.org/stable/2370790?seq=1. The second definition appears, for example, in Heath’s paper: https://www.ams.org/journals/proc/1961-012-05/S0002-9939-1961-0125562-9/S0002-9939-1961-0125562-9.pdf. It is unclear what the definition in Kundu and Garg’s paper, https://www.sciencedirect.com/science/article/pii/S0166864108003234 (the current citation for the property in pi-Base) is, since it is too vague and just says $X$ has a semimetric “compatible with its topology”, which really can be interpreted either way.

The second definition obviously implies the first. However, while the difference is subtle, the two definitions are, unfortunately, not equivalent. The first one more closely resembles the usual way metrizability is defined (and as such is more useful in practice, in my opinion), but it is quite weak: while it still implies $T_1$ and sequential, it *does not* imply first countable, while the second one does imply first countability, as noted in Heath’s paper.

I’m not sure whether adding two properties separately is a good idea, but otherwise a decision has to be made as to which version we should add.

**Comment by danflapjax** [comment]
@david20000813 That was a point of confusion for me when I first looked into the subject, so I appreciate you bringing it up.
That first definition is what is defined in *[Generalized Metric Spaces and Mappings](https://link.springer.com/book/10.2991/978-94-6239-216-8)* as *symmetrizable* whereas the second one is *semimetrizable*. They opt for the characterization that the neighborhood balls (i.e. $B(x,\varepsilon)=\\{y\in X\mid d(x,y)<\varepsilon\\}$) form a neighborhood base at each point.
I would like to add symmetrizable after adding semimetrizable, and I don't know of alternative set of terms that differentiate them.

**Comment by david20000813** [comment]
@danflapjax Ah, I wasn’t aware that there are terms available in the literature to differentiate the two. That’s nice and makes adding both properties workable.

**Comment by prabau** [comment]
It's not a problem to replace a theorem with one or more stronger ones once a new property is introduced.  We do it all the time. And it is preferable to have smaller PRs to make it easier to review and get merged.  Even for a single property, it is often better to split all the theorems into various batches. A gigantic PR is hard to review and runs the risk of languishing for a long time. (e.g. #821).

**Comment by david20000813** [comment]
@prabau That's good to know. Thanks

**Comment by prabau** [comment]
It would be good if this PR could mention that this is not the only definition in use.  Or @danflapjax, maybe you could add these clarifications later?

File: properties/P000102.md [hunk 1]
```diff
@@ -2,8 +2,20 @@
...
 
-Defined in Definition 3.7 of {{mr:2492954}}
+A space $X$ for which
+there is a distance function (or "semi-metric") $d:X^2\to[0,\infty)$
+satisfying reflexivity ($d(x,y)=0\Leftrightarrow x=y$) and symmetry
+($d(x,y)=d(y,x)$), such that if $p$ is
+a limit point of some $Y\subseteq X$, then $\inf\{d(y,p):y\in Y\}=0$.
+
+Defined in e.g. {{zb:0116.14303}}, where the following equivalence was
+given: the space is {P28} witnessed by $\{G_n(x):n<\omega\}$
```
Comment (danflapjax) [inline properties/P000102.md:16]:
```suggestion
given: if the space is {P28} witnessed by $\{G_n(x):n<\omega\}$
```
Apparently a function such as $G$ here is referred to as a $g$-function, but I'm not sure how much this is worth mentioning.

**Review by danflapjax — APPROVED** [review]

**Comment by danflapjax** [comment]
> It would be good if this PR could mention that this is not the only definition in use. Or @danflapjax, feel free to add a suggestion here, or it could also be clarified later.

I agree that it would be good to mention, but it would be also be easiest if we can just say "note that some sources, such as {source}, use the term semimetrizable to refer to {symmetrizable}", linking to the page on the latter.

**Comment by prabau** [comment]
> > It would be good if this PR could mention that this is not the only definition in use. Or @danflapjax, feel free to add a suggestion here, or it could also be clarified later.
> 
> I agree that it would be good to mention, but it would be also be easiest if we can just say "note that some sources, such as {source}, use the term semimetrizable to refer to {symmetrizable}", linking to the page on the latter.

Yeah, it can all be done later as part of introducing symmetrizable.

**Comment by david20000813** [comment]
Since semimetrizable is done, I can write a PR for adding symmetrizable, if no one objects? I’ll follow @danflapjax ’s suggestion and mention the different terminologies about “semimetrizable” in use, and add three theorems: “semimetrizable => symmetrizable”, “symmetrizable => $T_1$”, and “symmetrizable => sequential”.

**Pull request merged**